Update use of deprecatedProvideNotifications.

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #10465 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [ ] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

hussain-t commented on assets/js/components/adminbar/__snapshots__/AdminBarWidgets.test.js.snap:
```diff
@@ -89,6 +89,8 @@ exports[`AdminBarWidgets should not render the Activate Analytics CTA when the A
 </div>
 `;
 
+exports[`AdminBarWidgets should render the Admin Bar Widgets for the view only user if the module is shared 1`] = `<div />`;
+
```
Do you have any idea why this snapshot was updated?

hussain-t reviewed (changes requested):
Thanks, @ankitrox! This is almost good to go. However, a couple of concerns need to be addressed before we merge:
1. Please check my comment.
2. Remove the `deprecatedProvideNotifications` from `tests/js/utils.js`.

ankitrox commented:
Thanks for reviewing the PR @hussain-t.

I've addressed the feedback added in the comment. Assigning this to you to review the changes.

Thank you.

hussain-t reviewed (approved):
Nice work, @ankitrox! LGTM 👍
